src/revpi-factory-reset

```diff
@@ -103,34 +117,38 @@ echo "$hostname" > /etc/hostname
 
 # remove existing entries before writing new ones
 /bin/sed --follow-symlinks -r -i -e "/^dtparam=eth[0-9]_mac_/d" /boot/config.txt
-mac_hi="${mac:0:2}${mac:2:2}${mac:4:2}${mac:6:2}"
-mac_lo="${mac:8:2}${mac:10:2}"
-echo -e "MAC Address eth0:\t$mac_hi$mac_lo"
-echo "dtparam=eth0_mac_hi=0x${mac_hi}" >> /boot/config.txt
-echo "dtparam=eth0_mac_lo=0x${mac_lo}" >> /boot/config.txt
-if [[ "$ovl" =~ ^($witheth1_devicetype_list)$ ]] ; then
-	mac_lo1=$(( 0x${mac_lo} + 1))
-	mac_lo1=$(/usr/bin/printf "%.4hx" "${mac_lo1}")
-	echo -e "MAC Address eth1:\t$mac_hi$mac_lo1"
-	echo "dtparam=eth1_mac_hi=0x${mac_hi}" >> /boot/config.txt
-	echo "dtparam=eth1_mac_lo=0x${mac_lo1}" >> /boot/config.txt
-	if [[ "$ovl" =~ ^compact$ ]] ; then
-		/usr/sbin/ks8851-set-mac eth1 "$mac_hi$mac_lo1"
+if ! "$has_hat_eeprom"; then
+	mac_hi="${mac:0:2}${mac:2:2}${mac:4:2}${mac:6:2}"
+	mac_lo="${mac:8:2}${mac:10:2}"
+	echo -e "MAC Address eth0:\t$mac_hi$mac_lo"
+	echo "dtparam=eth0_mac_hi=0x${mac_hi}" >> /boot/config.txt
+	echo "dtparam=eth0_mac_lo=0x${mac_lo}" >> /boot/config.txt
+
+	if [[ "$ovl" =~ ^($witheth1_devicetype_list)$ ]] ; then
+		mac_lo1=$(( 0x${mac_lo} + 1))
+		mac_lo1=$(/usr/bin/printf "%.4hx" "${mac_lo1}")
+		echo -e "MAC Address eth1:\t$mac_hi$mac_lo1"
+		echo "dtparam=eth1_mac_hi=0x${mac_hi}" >> /boot/config.txt
+		echo "dtparam=eth1_mac_lo=0x${mac_lo1}" >> /boot/config.txt
+		if [[ "$ovl" =~ ^compact$ ]] ; then
+			/usr/sbin/ks8851-set-mac eth1 "$mac_hi$mac_lo1"
+		fi
 	fi
 fi
 
-if [[ "$ovl" =~ ^flat$ ]] ; then
+if ! "$has_hat_eeprom" && [[ "$ovl" =~ ^flat$ ]] ; then
```

Comment:
Why do you scatter the `if ! "$has_hat_eeprom"` all over the place? why not using one large block?